File: JavaLeetcodeProblems/src/1679.kt
class Solution1679 {
    fun maxOperations(nums: IntArray, k: Int): Int {
        nums.sort()

        var p1 = 0
        var p2 = nums.size - 1
        var count = 0

        while(p1 < p2) {
            if (nums[p1] + nums[p2] == k) {
                count++
                p1++
                p2--
            } else if (nums[p1] + nums[p2] < k) {
                p1++
            } else if (nums[p1] + nums[p2] > k) {
                p2--
            }
        }
        return count
    }
}


fun main() {

    //Test 1
    var input: IntArray = intArrayOf(1,2,3,4)
    var expected = 2
    var answer = Solution1679().maxOperations(input, 5)

    if (answer != expected) {
        println("Test 1: Failed (expected $expected, got $answer)")

    } else {
        println("Test 1: Passed")
    }


    //Test 2
    input = intArrayOf(3,1,3,4,3)
    expected = 1
    answer = Solution1679().maxOperations(input, 6)

    if (answer != expected) {
        println("Test 2: Failed (expected $expected, got $answer)")

    } else {
        println("Test 2: Passed")
    }
}

File: 75 Questions/1679.py
from typing import List

class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        nums.sort()
        count = 0

        if len(nums) == 0:
            return 0
        
        lp = 0
        rp = len(nums)-1

        while lp < rp:
            if nums[lp] + nums[rp] == k:
                count += 1
                lp += 1
                rp -= 1

            elif nums[lp] + nums[rp] < k:
                lp += 1

            else:
                rp -= 1

        return count